core/server/master/src/test/java/alluxio/master/file/MetadataSyncLockManagerTest.java

```diff
@@ -39,26 +38,44 @@ public class MetadataSyncLockManagerTest {
   public void setup() {
     Configuration.reloadProperties();
     Configuration.set(PropertyKey.MASTER_METADATA_SYNC_LOCK_POOL_INITSIZE, 0);
-    Configuration.set(PropertyKey.MASTER_METADATA_SYNC_LOCK_POOL_LOW_WATERMARK, 0);
-    Configuration.set(PropertyKey.MASTER_METADATA_SYNC_LOCK_POOL_HIGH_WATERMARK, 0);
     mMetadataSyncLockManager = new MetadataSyncLockManager();
   }
 
+  boolean differentIdentitiesFromIdx(int from, List<String> list1, List<String> list2) {
+    if (list1.size() != list2.size()) {
+      return false;
+    }
+    for (int i = from; i < list1.size(); i++) {
+      if (list1.get(i).equals(list2.get(i))) {
+        return false;
+      }
+    }
+    return true;
+  }
+
   @Test
-  public void lookPoolGC()
-      throws IOException, InterruptedException, TimeoutException, InvalidPathException {
+  public void lockPoolGC()
```

Comment:
I have a few more tests in mind:

1. Hold a lock to path `/a`. Trigger a GC, will the lock in the cache be gone? I think you need to make the YoungGen a bit crowded so there's really a reason to either reclaim the ref or upgrade it into the OldGen.
2. Hold a lock to path `/a`. Trigger some GC. If the expected behavior of ConcurrentMap is `/a` weak ref will be reclaimed, use another thread to acquire lock to `/a` and see it's the same or different lock. Note the old `/a` lock is never released. In this way we assert that through GCs, there will not be two different RWLock to `/a` which defeats concurrency control.

My understanding on the weakref map may not be 100% accurate, but I guess you get the spirit. If somehow a lock to a path is not found and results in two locks, then concurrency control is lost and that's bad. We should validate the behavior in the UT. In the old code everything is strong ref so there was no need.